Podverse/MakeClipTimeViewController.swift

```diff
@@ -113,27 +117,62 @@ class MakeClipTimeViewController: UIViewController, UITextFieldDelegate {
         }
         
         updateSpeedLabel()
+        loadMakeClipInputs()
+
+        if self.editingItem == nil {
+            self.title = "Make Clip"
+        } else {
+            self.title = "Edit Clip"
+        }
         
+        if (!hasSeenHint) {
+            animateHelper()
+        } else {
+            self.hintView.removeFromSuperview()
+            setupBarButtonItems()
+        }
+    }
+    
+    func animateHelper() {
+        if(self.shouldAnimate) {
+            if self.hintImageHorizontalConstraint != nil {
+                self.hintImageHorizontalConstraint.constant += 200
+                UIView.animate(withDuration: 1.5, animations: {
+                   self.view.layoutIfNeeded()
+                }) { (_) in
+                    if self.hintImageHorizontalConstraint != nil {
+                        self.hintImageHorizontalConstraint.constant -= 200
+                        UIView.animate(withDuration: 1.5, animations: {
+                            self.view.layoutIfNeeded()
+                        }) { (_) in
+                            self.animateHelper()
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    func setupBarButtonItems() {
         let saveBtn = UIBarButtonItem(title: "Save", style: .plain, target: self, action: #selector(save))
         let deleteBtn = UIBarButtonItem(title: "Delete", style: .plain, target: self, action: #selector(deleteClip))
         
         if self.editingItem == nil {
-            loadMakeClipInputs()
-            self.title = "Make Clip"
             self.navigationItem.rightBarButtonItems = [saveBtn]
         } else {
             // Make a copy in case the current PlayerHistoryItem reference disappears while on the edit view
             self.editingItem = self.editingItem?.copyPlayerHistoryItem()
-            loadEditClipInputs()
-            self.title = "Edit Clip"
             self.navigationItem.rightBarButtonItems = [saveBtn, deleteBtn]
         }
     }
     
     override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
         let touch:UITouch = touches.first!
-        if touch.view == self.grabHintImage {
-            self.grabHintImage.isHidden = true
+        if touch.view == self.hintView || touch.view == self.hintViewImage {
+            self.shouldAnimate = false
+            self.hintView.removeFromSuperview()
+            setupBarButtonItems()
+            UserDefaults.standard.set(true, forKey: "HAS_SEEN_CLIP_HINT")
         }
     }
```

Comment:
There's a weird effect happening right now where, when the hint view loads, the elements behind it slowly animate into their positions as well. Have you seen this? Do you know what's causing it? I'm thinking it'd look better without it.

Reply:
Its the layoutIfNeeded call that happens for to the animation. It gets called inside an animate block so any change in frame actually animates. I can remove it but we'll loose the little hand animation